File: Lab 7/t1.c
#include <stdio.h>

// "the total number of integers will not exceed 50"
int a[50] = {0};

int main (void)
{
  int r = 0;
  int count = 0; // Count number of numbers input by user

  puts ("Please input some numbers (input 0 to end):"); /* exit):"); */
  for (count = 0; count < 50; count++) {
    r = scanf ("%d", &a[count]);
    if (r != 1) {
      puts ("Illegal input detected, exiting.");
      return 1;
    }
    if (a[count] == 0) {
      break;
    }
  }

  puts ("Among your inputs, the following are positive:");
  for (int i = 0; i < count; i++) {
    if (a[i] > 0) {
      printf("%d ", a[i]);
    }
  }
  putchar ('\n');

  return 0;
}

File: Lab 7/t1.go
package main

import (
  "fmt"
  "os"
  "bufio"
)

func main() {
  os.Exit(t1_main())
}

func t1_main() int {
  var a [50]int
  var count int
  var e error

  fmt.Println("Please input some numbers (input 0 to end):")
  stdin := bufio.NewReader(os.Stdin)
  for count = 0; count < 50; count += 1 {
    _, e = fmt.Fscanf(stdin, "%d", &a[count])
    if e != nil {
      fmt.Println("Illegal input detected, exiting.")
      return 1
    }
    if a[count] == 0 {
      break
    }
  }

  fmt.Println("Among your inputs, the following are positive:")
  for i := 0; i < count; i += 1 {
    if a[i] > 0 {
      fmt.Print(a[i], " ")
    }
  }
  fmt.Print("\n")

  return 0
}
